sdk/storage/inc/datalake/path_client.hpp

```diff
@@ -0,0 +1,249 @@
+     *             control rights that were present earlier on files and directories
+     * @param options Optional parameters to set an access control recursively to the resource the
+     *                path points to.
+     * @return PathSetAccessControlRecursiveResponse
+     */
+    PathSetAccessControlRecursiveResponse SetAccessControlRecursive(
+        PathSetAccessControlRecursiveMode mode,
+        const SetAccessControlRecursiveOptions& options = SetAccessControlRecursiveOptions()) const;
+
+    /**
+     * @brief Sets the properties of a resource the path points to.
+     * @param options Optional parameters to set the properties to the resource the path points to.
+     * @return PathUpdateResponse
+     */
+    PathUpdateResponse SetProperties(
+        const SetPathPropertiesOptions& options = SetPathPropertiesOptions()) const;
+
+    /**
+     * @brief Creates a file or directory. By default, the destination is overwritten and
+     *        if the destination already exists and has a lease the lease is broken.
+     * @param options Optional parameters to create the resource the path points to.
+     * @return PathCreateResponse
+     */
+    PathCreateResponse Create(const PathCreateOptions& options = PathCreateOptions()) const;
+
+    /**
+     * @brief Renames a file or directory. By default, the destination is overwritten and
+     *        if the destination already exists and has a lease the lease is broken.
+     * @param options Optional parameters to rename a resource to the resource the path points to.
+     * @return PathRenameResponse
+     */
+    PathRenameResponse Rename(const PathRenameOptions& options = PathRenameOptions()) const;
+
+    /**
+     * @brief Deletes the file or directory.
+     * @param options Optional parameters to delete the resource the path points to.
+     * @return PathDeleteResponse
+     */
+    PathDeleteResponse Delete(const PathDeleteOptions& options = PathDeleteOptions()) const;
+
+    /**
+     * @brief Get Properties returns all system and user defined properties for a path. Get Status
+     *        returns all system defined properties for a path. Get Access Control List returns the
+     *        access control list for a path.
+     * @param options Optional parameters to get the properties from the resource the path points
+     *                to.
+     * @return GetPathPropertiesResponse
+     */
+    GetPathPropertiesResponse GetProperties(
+        const PathGetPropertiesOptions& options = PathGetPropertiesOptions()) const;
+
+    /**
+     * @brief Create and manage a lease to restrict write and delete access to the path.
+     * @param options Optional parameters to create or manage a lease on the resource the path
+     *                points to.
+     * @return PathLeaseResponse
+     */
+    PathLeaseResponse Lease(const PathLeaseOptions& options = PathLeaseOptions()) const;
```

Comment:
I think we already reached an agreement that we wouldn't add lease functions until we had a design accepted by all.